python/lsst/summit/testing/analysis/m1m3/plots/inertia_compensation_system.py

```diff
@@ -0,0 +1,272 @@
+import matplotlib.pyplot as plt
+import pandas as pd
+from astropy.time import Time
+
+
+def plot_hp_data(ax: plt.Axes, data: pd.Series | list, label: str) -> list[plt.Line2D]:
+    """
+    Plot hardpoint data on the given axes.
+
+    Parameters
+    ----------
+    ax : matplotlib.axes._axes.Axes
+        The axes on which the data is plotted.
+    topic : str
+        The topic of the data.
+    data : Series or list
+        The data points to be plotted.
+    label : str
+        The label for the plotted data.
+
+    Returns
+    -------
+    list
+        A list containing the Line2D objects representing the plotted data
+        lines.
+    """
+    line = ax.plot(data, "-", label=label, lw=0.5)
+    return line
+
+
+def mark_slew_begin_end(ax: plt.Axes, slew_begin: Time, slew_end: Time) -> plt.Line2D:
+    """
+    Mark the beginning and the end of a slew with vertical lines on the given
+    axes.
+
+    Parameters
+    ----------
+    ax : matplotlib.axes._axes.Axes
+        The axes where the vertical lines are drawn.
+    slew_begin : astropy.time.Time
+        The slew beginning time.
+    slew_end : astropy.time.Time
+        The slew ending time.
+
+    Returns
+    -------
+    matplotlib.lines.Line2D
+        The Line2D object representing the line drawn at the slew end.
+    """
+    _ = ax.axvline(slew_begin.datetime, lw=0.5, ls="--", c="k", zorder=-1)
+    line = ax.axvline(
+        slew_end.datetime, lw=0.5, ls="--", c="k", zorder=-1, label="Slew Start/Stop"
+    )
+    return line
+
+
+def mark_padded_slew_begin_end(ax: plt.Axes, begin: Time, end: Time) -> plt.Line2D:
+    """
+    Mark the padded beginning and the end of a slew with vertical lines.
+
+    Parameters
+    ----------
+    ax : matplotlib.axes._axes.Axes
+        The axes where the vertical lines are drawn.
+    begin : astropy.time.Time
+        The padded slew beginning time.
+    end : astropy.time.Time
+        The padded slew ending time.
+
+    Returns
+    -------
+    matplotlib.lines.Line2D
+        The Line2D object representing the line drawn at the padded slew end.
+    """
+    _ = ax.axvline(begin.datetime, alpha=0.5, lw=0.5, ls="-", c="k", zorder=-1)
+    line = ax.axvline(
+        end.datetime,
+        alpha=0.5,
+        lw=0.5,
+        ls="-",
+        c="k",
+        zorder=-1,
+        label="Padded Slew Start/Stop",
+    )
+    return line
+
+
+def customize_hp_plot(ax: plt.Axes, dataset: object, lines: list[plt.Line2D]) -> None:
```

Comment:
I think you are having issues with mypy because you are using `dataset: object` . If you don't know the type, you should probably use `typing.Any` instead though. 

However,  you could consider creating an interface definition for this object. It would be something like this:

```
from typing import Protocol

class Event(Protocol):
    @property
    def dayObs(self) -> int:
        """Day of the observation." 

    @property
    def seqNum(self) -> int:
        """Day of the observation." 

    # define everything you call from event here.
    ...


class DataSet(Protocol):
    @property
    def event(self) -> Event:
        """Event"""
```

Then here you would do:

```
def customize_hp_plot(ax: plt.Axes, dataset: DataSet, lines: list[plt.Line2D]) -> None:
```


I would define those protocols in a `type_utils` submodule.

Reply:
I know that each dataset should be a M1M3ICSAnalysis. At some point, I was confused about whether to import it from the upper-level module or leave it as an object. I'll try the first approach because it seems simpler. If not, I will apply your suggestion of using Protocol.